pymbolic/mapper/equalizer.py

```diff
@@ -0,0 +1,169 @@
+import pymbolic
```

Comment:
Add an `__eq__` dispatch that asks both `self` and `other` for `make_equality_comparer`. Make this fail if the two don't agree on a  class (`issubclass` one way or the other).